### PR TITLE
fix: Item overlay rendering relating to charge count

### PR DESCRIPTION
**Pull request by ryanzhoudev**
- Added a check to the charge count overlay to see if items are actually in the hotbar
- Reverted ItemLevelOverlay to previous change to ensure level overlay hotkey still works with hotbar overlay config on
- Separated hotbar charge overlay to its own file to improve check logic

### CODE REVIEW

**Comment by ryanzhoudev** [comment]
Typo fixes is actually kind of questionable, I think that resets configs? Not sure

Edit - yes it does, pretend that never happened

File: src/main/java/com/wynntils/modules/utilities/overlays/inventories/HotbarChargeOverlay.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.text.TextFormatting;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+
+public class HotbarChargeOverlay implements Listener {
+
+    @SubscribeEvent
+    public void onHotbarOverlay(RenderEvent.DrawItemOverlay event) {
+        if (!UtilitiesConfig.INSTANCE.showConsumableChargesHotbar) return;
+        ItemStack is = event.getStack();
```
Comment (kristofbolyai) [inline src/main/java/com/wynntils/modules/utilities/overlays/inventories/HotbarChargeOverlay.java]: Rename this variable for clarity

File: src/main/java/com/wynntils/modules/utilities/overlays/inventories/HotbarChargeOverlay.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        Item item = is.getItem();
+
+        // We don't care about items inside the inventory, only in the hotbar
+        if (McIf.player().inventory.mainInventory.contains(is) && McIf.player().inventory.mainInventory.indexOf(is) > 8) return;
```
Comment (kristofbolyai) [inline src/main/java/com/wynntils/modules/utilities/overlays/inventories/HotbarChargeOverlay.java]: Does this actually work with all inventories?

**Review by kristofbolyai — DISMISSED** [review]

**Comment by kristofbolyai** [comment]
Did not mean to approve just yet.

File: src/main/java/com/wynntils/modules/utilities/overlays/inventories/HotbarChargeOverlay.java (hunk 2 above)
Comment (kristofbolyai) [reply src/main/java/com/wynntils/modules/utilities/overlays/inventories/HotbarChargeOverlay.java]: It think it does, nevermind.

File: src/main/java/com/wynntils/modules/utilities/overlays/inventories/HotbarChargeOverlay.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        if (!(item == Items.POTIONITEM) || item == Items.DIAMOND_AXE) return; // Consumables are only potions or diamond axes (crafted)
```
Comment (kristofbolyai) [inline src/main/java/com/wynntils/modules/utilities/overlays/inventories/HotbarChargeOverlay.java]:
```suggestion
        if (item != Items.POTIONITEM && item != Items.DIAMOND_AXE) return; // Consumables are only potions or diamond axes (crafted)
```

Isn't this what you meant to do?

**Review by kristofbolyai — CHANGES_REQUESTED** [review]

File: src/main/java/com/wynntils/modules/utilities/overlays/inventories/HotbarChargeOverlay.java (hunk 3 above)
Comment (ryanzhoudev) [reply src/main/java/com/wynntils/modules/utilities/overlays/inventories/HotbarChargeOverlay.java]: Yup thanks lol, I just copied directly from the previous file without really looking at it too much

**Review by kristofbolyai — APPROVED** [review]

**Review by magicus — APPROVED** [review]

**Pull request merged**